website/src/lapis/getCladeLineages.ts

```diff
@@ -0,0 +1,72 @@
+import axios from 'axios';
+import { z } from 'zod';
+
+import { getClientLogger } from '../clientLogger.ts';
+
+const logger = getClientLogger('getCladeLineages');
+
+const aggregatedSchema = z.object({
+    data: z.array(
+        z
+            .object({
+                count: z.number(),
+            })
+            .catchall(z.union([z.string(), z.number(), z.boolean(), z.null()])),
+    ),
+});
+
+/**
+ * Finds the lineage definition belonging to clades, by looking for the lineage that is most
+ * commonly associated with each clade.
+ * Returns a map, mapping all clades found in the data to their respective lineage.
+ */
+export async function getCladeLineages(
+    baseUrl: string,
+    cladeField: string,
+    lineageField: string,
+): Promise<Record<string, string>> {
+    const url = `${baseUrl.replace(/\/$/, '')}/sample/aggregated`;
+    const body = {
+        fields: [cladeField, lineageField],
+        orderBy: [cladeField],
+    };
+
+    let response;
+    try {
+        response = await axios.post(url, body);
+    } catch (error) {
+        const message = `Failed to fetch clade lineages: ${JSON.stringify(error)}`;
+        logger.error(message);
+        throw new Error(message);
+    }
+
+    const parsedResponse = aggregatedSchema.safeParse(response.data);
+    if (!parsedResponse.success) {
+        const message = `Failed to parse clade lineages response: ${JSON.stringify(parsedResponse)} (was ${JSON.stringify(response.data)})`;
+        logger.error(message);
+        throw new Error(message);
+    }
+
+    const mapping: Record<string, { lineage: string; count: number }> = {};
+
+    for (const row of parsedResponse.data.data) {
+        const clade = row[cladeField];
+        const lineage = row[lineageField];
+        const count = row.count;
+
+        if (clade === null || lineage === null) {
+            continue;
+        }
+
+        if (typeof clade !== 'string' || typeof lineage !== 'string' || typeof count !== 'number') {
```

Comment:
```suggestion
        if (typeof clade !== 'string' || typeof lineage !== 'string') {
```

That should have already been validated by the schema.